static/js/controllers/contacts-content.js

```diff
@@ -211,29 +213,37 @@ var _ = require('underscore'),
         });
       };
 
-      var removePrimaryContact = function(children, primaryContactDoc) {
-        if (!children || children.length === 0 || !primaryContactDoc) {
-          return;
-        }
-        var primaryContactIdx = _.findIndex(children, function(child) {
-          return child.doc._id === primaryContactDoc._id;
-        });
-        if (primaryContactIdx < 0) {
-          return;
+      var placePrimaryContactOnTop = function(primaryContactDoc, persons) {
+        var removePrimaryContact = function(children, primaryContactDoc) {
+          if (!children || children.length === 0 || !primaryContactDoc) {
+            return;
+          }
+          var primaryContactIdx = _.findIndex(children, function(child) {
+            return child.doc._id === primaryContactDoc._id;
+          });
+          if (primaryContactIdx < 0) {
+            return;
+          }
+          children.splice(primaryContactIdx, 1);
+        };
+
+        if (primaryContactDoc) {
+          removePrimaryContact(persons, primaryContactDoc);
+          persons.unshift({ id: primaryContactDoc._id, doc: primaryContactDoc, isPrimaryContact: true });
         }
-        children.splice(primaryContactIdx, 1);
       };
 
-      var sortChildPersons = function(persons, contactDoc, primaryContactDoc) {
-        if (persons.length) {
+
+      var sortChildPersons = function(children, contactDoc, primaryContactDoc) {
```

Comment:
If you don't want to pass `children` (you used to pass `persons`) you can have `sortChildPersons` return the sorted list instead. IMO it's better anyway, since magical state changing isn't happening in a function, but that's entirely personal preference.